library/common/http/client.h

```diff
@@ -42,11 +43,13 @@ struct HttpClientStats {
 class Client : public Logger::Loggable<Logger::Id::http> {
 public:
   Client(ApiListener& api_listener, Event::ProvisionalDispatcher& dispatcher, Stats::Scope& scope,
-         std::atomic<envoy_network_t>& preferred_network, Random::RandomGenerator& random)
+         std::atomic<envoy_network_t>& preferred_network, Random::RandomGenerator& random,
+         bool explicit_buffering = false)
```

Comment:
I realize there are no API docs previously, but can you add some comments on what "explicit buffering" is? It's not clear to from a quick skim what this parameter does.

Edit: after reading I think I understand what this does. I might actually call this `explicit_flow_control`? IMO that would be a bit more clear (along with some API docs)? WDYT?

Reply:
@mattklein123, we had called it `flow_control_enabled` previously, but @RyanTheOptimist noted that that could be misinterpreted as enabling/disabling HTTP flow control. I actually think `explicit_flow_control` is an improvement because it maybe carries less of that implication, while "flow control" still somewhat better captures the intent.

Reply:
Yeah I have angst about flow control being an ambiguous term in this space. That being said, none of the names are ideal and I agree with @goaway that `explicit_flow_control` is an improvement from `flow_control_enabled`, so it works for me.